code/modules/hydroponics/grown.dm

```diff
@@ -72,15 +86,93 @@
 		return
 	return
 
+
+// Various gene procs
+/obj/item/weapon/reagent_containers/food/snacks/grown/attack_self(mob/user)
+	if(seed.get_gene(/datum/plant_gene/trait/squash))
+		squash(user)
+	..()
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/throw_impact(atom/hit_atom)
+	if(!..() && seed.get_gene(/datum/plant_gene/trait/squash)) //was it caught by a mob?
+		squash(hit_atom)
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/proc/squash(atom/target)
+	var/turf/T = get_turf(target)
+	if(ispath(splat_type, /obj/effect/decal/cleanable/plant_smudge))
+		if(filling_color)
+			var/obj/O = new splat_type(T)
+			O.color = filling_color
+			O.name = "[name] smudge"
+	else if(splat_type)
+		new splat_type(T)
+
+	if(trash)
+		if(ispath(trash, /obj/item/weapon/grown) || ispath(trash, /obj/item/weapon/reagent_containers/food/snacks/grown))
+			new trash(T, seed)
+		else
+			new trash(T)
+
+	visible_message("<span class='warning'>[src] has been squashed.</span>","<span class='italics'>You hear a smack.</span>")
+	for(var/datum/plant_gene/trait/trait in seed.genes)
+		trait.on_squash(src, target)
+
+	for(var/atom/A in T)
+		reagents.reaction(A)
+
+	qdel(src)
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/On_Consume()
+	if(iscarbon(usr))
+		for(var/datum/plant_gene/trait/T in seed.genes)
+			T.on_consume(src, usr)
+	..()
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/Crossed(AM as mob|obj)
+	var/datum/plant_gene/trait/slip/S = seed.get_gene(/datum/plant_gene/trait/slip)
+	if(S && !ispath(trash, /obj/item/weapon/grown) && istype(AM, /mob/living/carbon))
+		var/mob/living/carbon/M = AM
+		var/stun = min(seed.potency * S.rate * 2, 1)
+		var/weaken = min(seed.potency * S.rate, 0.5)
+		if(M.slip(stun, weaken, src))
+			for(var/datum/plant_gene/trait/T in seed.genes)

```

Comment:
Fuck no. `genes` contains different gene types now, it will contain even more types in the future, and I'm not going to split it into the different vars just because some bug-alien wants me to do it.